packages/shared/src/models-vis.ts

```diff
@@ -36,3 +36,7 @@ export interface Dims {
   rows: number;
   cols: number;
 }
+
+// MappedTupple<number[], string> => string[]
+// MappedTupple<[number, number], string> => [string, string]
+export type MappedTuple<T, U> = { [index in keyof T]: U };
```

Comment:
This is the only slightly tricky bit, really. Hopefully the commented examples are self-explanatory.